### DISCUSSION TITLE
Bulk action for automatic refunds

### DISCUSSION THREAD

**Discussion by jeriox** [pretix/pretix#3699] [Ideas]
We have a lot of orders (>200) with cancelled positions, which means they are overpaid. Pretix already hints that on the overview and one can view them at `/orders/?status=overpaid`. For every single order, the process to issue a refund is fine (navigating to the order detail page, clicking the "refund overpaid amount" button in the alert, having the amount prefilled in the refund form for the payment method that was used and just being able to click "perform refund" without the need for further inputs). However, doing so for a large amount of orders is tedious. As the order list already allows for bulk action, we propose a refund bulk action that tries to automatically refund the overpaid amount and fails if that is impossible (e.g. because the payment provider does not support it).

We are currently solving this problem by POSTing to the `/payments/(local_id)/refund/` API endpoint, but that only works because all orders have the same refund amount, as the overpaid value is not part of the order endpoint. Doing so in the webinterface would be much nicer.

We would be willing to provide an implementation for the bulk action if you agree that it would be helpful.

**Comment by raphaelm** [comment]
> As the order list already allows for bulk action, we propose a refund bulk action that tries to automatically refund the overpaid amount and fails if that is impossible (e.g. because the payment provider does not support it).

I think that could work. We intentionally did not add bulk-cancel in that place since it is dangerous:tm:, but for refunding already-overpaid amounts I think it's manageable risk. Happy to look at a PR for it if you're offering :)

> We are currently solving this problem by POSTing to the /payments/(local_id)/refund/ API endpoint, but that only works because all orders have the same refund amount, as the overpaid value is not part of the order endpoint.

Not explicitly, but it is the order total minus the sum of all payments plus the sum of all refunds, so it's also not too hard to compute in an API client.
